elements/lisk-transaction-pool/src/transaction_pool.ts

```diff
@@ -217,6 +215,29 @@ export class TransactionPool {
 			);
 		}
 
+		// Add the PROCESSABLE, UNPROCESSABLE transaction to _transactionList and set PROCESSABLE as true
+		const { added, removedID } = this._transactionList[incomingTxAddress].add(
+			incomingTx,
+			txStatus === TransactionStatus.PROCESSABLE,
+		);
+
+		if (removedID) {
+			debug('Removing from transaction pool with id', removedID);
+			delete this._allTransactions[removedID];
+		}
+
+		if (!added) {
```

Comment:
Maybe we can move this condition before the above condition and early return error